src/pubtools/_marketplacesvm/tasks/delete/command.py

```diff
@@ -0,0 +1,271 @@
+
+    def _convert_provider_name(self, provider_name: str) -> List[str]:
+        # Since we're only focused on AWS these are the accounts we're dealing with.
+        # AmiProduct is a list as there's no way of telling which account the image
+        # is on until we try to delete it.
+        # Eventually we might want to add to pushsource a provider name field.
+        accounts = {
+            "AWS": ["aws-us-storage"],
+            "AGOV": ["aws-us-gov-storage"],
+            "ACN": ["aws-china-storage"],
+            "AmiProduct": ["aws-na", "aws-emea"],
+        }
+        return accounts[provider_name]
+
+    def _delete(
+        self,
+        push_item: VMIPushItem,
+        **kwargs,
+    ) -> VMIPushItem:
+        builds = self.args.builds
+        dry_run = self.args.dry_run
+        keep_snapshots = self.args.keep_snapshots
+        marketplaces = self._convert_provider_name(push_item.marketplace_entity_type)
+        for marketplace in marketplaces:
+            if dry_run:
+                log.info("Would have deleted: %s in build %s", push_item.image_id, push_item.build)
+                self._SKIPPED = True
+                pi = evolve(push_item, state=State.SKIPPED)
+            elif push_item.build in builds:
+                log.info(
+                    "Deleting %s in account %s",
+                    push_item.image_id,
+                    marketplace,
+                )
+                try:
+                    pi = self.cloud_instance(marketplace).delete(
+                        push_item, keep_snapshots=keep_snapshots, **kwargs
+                    )
+                    log.info(
+                        "Delete finished for %s in account %s",
+                        push_item.image_id,
+                        marketplace,
+                    )
+                except Exception as exc:
+                    # If we failed the image might not exist, not necessarily an error
+                    log.info(
+                        "Failed to delete %s in %s:%s",
+                        push_item.image_id,
+                        marketplace,
+                        str(exc),
+                        stack_info=True,
+                    )
+                    self._SKIPPED = True
+                    pi = evolve(push_item, state=State.SKIPPED)
+                    return pi
+            else:
+                log.info("Skipped: %s in build %s", push_item.image_id, push_item.build)
+                self._SKIPPED = True
+                pi = evolve(push_item, state=State.SKIPPED)
+                return pi
+
+        try:
+            self.update_rhsm_metadata(push_item, marketplace)
+        except Exception as exc:
+            log.info(
+                "Failed to contact RHSM for %s in %s:%s",
+                push_item.image_id,
+                marketplace,
+                str(exc),
+                stack_info=True,
+            )
```

Comment:
IIUC this will happily fail as a `log.info` instead of `log.error` or `log.raise` and mark the task as successfull which s dangerous..

Please make sure it re-raises and the log is at least in error

Reply:
Fixed a lot of these issues. If you have time to look it back over :)